ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -449,9 +452,59 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, err
 			}
 		}
+		if rayJobInstance.Spec.Schedule != "" {
+			logger.Info("RayJob is scheduled again")
+			rayJobInstance.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusScheduling
+			break
+		}
 
 		// If the RayJob is completed, we should not requeue it.
 		return ctrl.Result{}, nil
+	case rayv1.JobDeploymentStatusScheduling:
+		deleteCluster := rayJobInstance.Spec.ShutdownAfterJobFinishes
```

Comment:
nit: this variable is only called once, we can remove this line and just check `rayJobInstance.Spec.ShutdownAfterJobFinishes` directly on line 477